typing/typedecl_variance.ml

```diff
@@ -122,7 +122,7 @@ let compute_variance_type env ~check (required, loc) decl tyl =
     List.map
       (fun (c,n,i) ->
         let i = if check_injectivity then i else false in
-        if c || n then (c,n,i) else (true,true,i))
+        if c <> n then (c,n,i) else (not c,not n,i))
```

Comment:
This line seems to be the most tricky change.
Jacques and I could not find an immediate way to refactor this part.

Reply:
To be more precise, c and n in required reflect + and - in the syntax, not the corresponding variance bits, do a conversation is needed here. 
It would be nicer to do the conversation earlier, but other parts of the code depend on this meaning, so this should be a separate refactoring.

Reply:
👍
Thank you so much for a careful reading and a nice cleaning.